### ISSUE TITLE
set_mark documentation

### ISSUE THREAD

**Issue by joshgoebel** [joshgoebel/xkeysnail#8]
Ref: https://github.com/mooz/xkeysnail/issues/97

> I'm not sure what set_mark etc are supposed to do, I can make a PR for the README out of your answer here if that would help!

@jackmac92 If you'd be interested in making a PR here to document this I'd be happy to accept it.  Also we have the wiki available though I haven't decided for sure how we should use it yet...  as things stand now the README makes sense.

I plan on maintaining this fork (and seeking other contributors) since mooz seems to not be active anymore with xkeysnail.

**Comment by joshgoebel** [comment]
Personally I find `with_or_set_mark` quite confusing and wonder if we need to rename it...

**Comment by ellakk** [comment]
> Personally I find `with_or_set_mark` quite confusing and wonder if we need to rename it...

Hi,
I wrote that code along with the multipurpose key support for xkeysnail many years ago. The purpose behind it is for xkeysnail to emulate `C-M-SPC` (from Emacs) to rapidly mark multiple words in a row, might not be useful for anything else. 

On another note, great job you guys have done here! I've been using xkeysnail from the start and keysnail for Firefox before that, but I never noticed this fork until now. Going to start to migrate over as soon as I get some time :)

**Comment by joshgoebel** [comment]
Yeah I wasn't questioning the utility, just the name.  Is this a named concept directly from Emacs that I'm just not familiar with? ("with or set mark")  or just your attempt to name something?

**Comment by ellakk** [comment]
Sorry, it is indeed Emacs terminology. Like [lenbook](https://github.com/mooz/xkeysnail/issues/97#issuecomment-665408971) mentioned, it's about the state of the [mark](https://www.gnu.org/software/emacs/manual/html_node/emacs/Mark.html). Named like an Emacs `with`-macro that is something you wrap around code to ensure it's executed under specific conditions. It should ideally been called `with-mark` to indicate that the wrapped keys are executed with the mark set, but it was taken.

**Comment by joshgoebel** [comment]
> but it was taken.

Do you have any better name suggestions?  If we had much clearer names I'd be ok changing them and breaking old configs in exchange for a lot more clarity.  Perhaps it'd help if you explain the emacs concept briefly?  Or perhaps share the emacs-like portion of your own config file so I could see how you're using all these different helper in real combos?
